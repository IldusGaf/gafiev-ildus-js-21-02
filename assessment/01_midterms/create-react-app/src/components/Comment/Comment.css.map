{"version":3,"sources":["Comment.less"],"names":[],"mappings":"AAAA;EACI,aAAA;EACA,gBAAA;EACA,gBAAA;EACA,gBAAA;EACA,aAAA;EACA,aAAA;EACA,SAAA;EACA,eAAA;EACA,YAAA;EACA,iBAAA;EACA,mBAAA;;AAXJ,QAaI;EACI,WAAA;EACA,aAAA;EACA,sBAAA;;AAhBR,QAaI,oBAKI;EACI,WAAA;EACA,aAAA;EACA,8BAAA;EACA,mBAAA;;AAtBZ,QAaI,oBAKI,iBAMI;EACI,eAAA;;AAMhB;EACI,mBAAA","file":"Comment.css"}