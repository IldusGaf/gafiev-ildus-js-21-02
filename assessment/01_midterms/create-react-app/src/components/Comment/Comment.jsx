import React from 'react';
import './Comment.css';
import { Avatar } from 'antd';
import ComponentWithHelper from '../../wrappers/ComponentWithHelper/ComponentWithHelper';
import { Link } from 'react-router-dom';

const Comment = (props) => {
  return (
    <div className={`comment ${props.darkTheme && 'comment_dark'}`}>
      <Avatar shape="circle" src={props.avatar} />
      <div className="comment__container">
        <div className="comment__header">
          <ComponentWithHelper text={props.id}>
            <div className="comment__owner">
              <Link to={`/users/${props.id}`}>
                {props.title
                  ? `${props.title} ${props.firstName} ${props.lastName}`
                  : `${props.firstName} ${props.lastName}`}
              </Link>
            </div>
          </ComponentWithHelper>
          <div className="comment__date">{props.publishDate && props.publishDate.slice(0, 10)}</div>
        </div>
        <div className="comment__text">{props.text}</div>
      </div>
    </div>
  );
};

export default Comment;
