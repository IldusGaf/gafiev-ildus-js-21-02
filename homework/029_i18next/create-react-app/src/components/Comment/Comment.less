.comment {
    flex: 1 1 45%;
    max-width: 450px;
    min-width: 250px;
    min-height: 70px;
    padding: 10px;
    display: flex;
    gap: 10px;
    font-size: 12px;
    color: black;
    background: #ffff;
    border-radius: 10px;

    .comment__container {
        width: 100%;
        display: flex;
        flex-direction: column;

        .comment__header {
            width: 100%;
            display: flex;
            justify-content: space-between;
            align-items: center;

            .comment__owner {
                font-size: 14px;
            }
        }
    }
}

.comment_dark {
    background: #f2f2f2;
}