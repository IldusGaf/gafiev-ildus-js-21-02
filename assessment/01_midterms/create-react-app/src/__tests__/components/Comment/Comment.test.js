import React from "react";
import { shallow } from 'enzyme';
import Comment from "../../../components/Comment/Comment";

describe('Comment component', () => {
    test('should render component', () => {
        const wrapper = shallow(<Comment />);
        expect(wrapper.find('div.comment')).toHaveLength(1);
    });

    test('snapshot test', () => {
        const wrapper = shallow(<Comment />);
        expect(wrapper).toMatchSnapshot();
      });

    test('should render data', () => {
        const title = 'mr';
        const firstName = 'James';
        const lastName = 'Bond';
        const publishDate = '12.06.2020';
        const text = 'Blah-blah-blah';
        const wrapper = shallow(<Comment
           title={title}
           firstName={firstName} 
           lastName={lastName}
           publishDate={publishDate}
           text={text}
        />);
        expect(wrapper.find('div.comment__owner').first().text()).toBe(`${title} ${firstName} ${lastName}`);
        expect(wrapper.find('div.comment__date').text()).toBe(publishDate);
        expect(wrapper.find('div.comment__text').text()).toBe(text);
    });
});  
