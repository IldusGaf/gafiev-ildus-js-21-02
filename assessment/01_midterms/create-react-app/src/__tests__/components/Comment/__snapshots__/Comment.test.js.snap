// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Comment component snapshot test 1`] = `
<div
  className="comment undefined"
>
  <Avatar
    shape="circle"
    size="default"
  />
  <div
    className="comment__container"
  >
    <div
      className="comment__header"
    >
      <ComponentWithHelper>
        <div
          className="comment__owner"
        >
          <Link
            to="/users/undefined"
          >
            undefined undefined
          </Link>
        </div>
      </ComponentWithHelper>
      <div
        className="comment__date"
      />
    </div>
    <div
      className="comment__text"
    />
  </div>
</div>
`;
